File: Python/XOREncryption.py
#!/usr/bin/env python
# encoding: utf-8
"""
XOREncryption.py

Created by Kyle Banks on 2013-10-06.
"""

def encryptDecrypt(input):
	key = ['K', 'C', 'Q'] #Can be any chars, and any size array
	output = []
	
	for i in range(len(input)):
		xor_num = ord(input[i]) ^ ord(key[i % len(key)])
		output.append(chr(xor_num))
	
	return ''.join(output)


def main():
	encrypted = encryptDecrypt("kylewbanks.com");
	print("Encrypted:"+encrypted);
	
	decrypted = encryptDecrypt(encrypted);
	print("Decrypted:"+decrypted);
	pass


if __name__ == '__main__':
	main()

File: Kotlin/XOREncryption.kt
fun encryptDecrypt(input: String): String {
    val key = charArrayOf('K', 'C', 'Q') //Can be any chars, and any length array
    val output = StringBuilder()

    for (i in 0 until input.length) {
        val a: Int = input[i].toInt()
        val b: Int = key[i % key.size].toInt()
        output.append((a xor b).toChar())
    }

    return output.toString()
}

fun main(args: Array<String>) {
    val encrypted = encryptDecrypt("kylewbanks.com")
    println("Encrypted: $encrypted") // output: Encrypted:  :=.43*-:8m2$.

    val decrypted = encryptDecrypt(encrypted)
    println("Decrypted: $decrypted") // output: Decrypted: kylewbanks.com
}